packages/core/src/App/Containers/Layout/header/traders-hub-onboarding.tsx

```diff
@@ -25,10 +27,17 @@ const TradersHubOnboarding = observer(() => {
                         data_testid='dt_traders_hub_onboarding_icon'
                         icon={is_dark_mode_on ? 'IcAppstoreTradingHubOnboardingDark' : 'IcAppstoreTradingHubOnboarding'}
                         size={20}
-                        onClick={() => {
-                            history.push(routes.onboarding);
-                            setIsOnboardingVisited(false);
-                        }}
+                        onClick={
+                            is_next_wallet_enabled
+                                ? () => {
+                                      localStorage.setItem('walletsOnboarding', 'started');
+                                      window.dispatchEvent(new Event('storage'));
```

Comment:
can we avoid dispatching events globally somehow? it seems like this is used effectively to let other component observe changes in this component, which generally should be avoided (unless its necessary or really hard to avoid)

Reply:
@wojciech-deriv I need `window.dispatchEvent` because `eventListener` without event dispatching can catch `storage` event only from other browser tabs (or when I change it manually)